pyramid/request.py

```diff
@@ -214,10 +213,8 @@ def response(self):
         right" attributes (e.g. by calling ``request.response.set_cookie()``)
         within a view that uses a renderer.  Mutations to this response object
         will be preserved in the response sent to the client."""
-        registry = self.registry
-        response_factory = registry.queryUtility(IResponseFactory,
-                                                 default=Response)
-        return response_factory()
+        response_factory = _get_response_factory(self.registry)
+        return response_factory(self)
```

Comment:
I think maybe what you are referring to is changing the factory to accept the `request` object. Unfortunately `IResponseFactory` was a private API prior to 1.6 so we do not document those changes.